CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -802,44 +853,33 @@ local function CreateSettingsIcon(topBarInstance)
 	return menuItem
 end
 
-local function Create3DSettingsIcon(topBarInstance, panel)
-	local menuItem = CreateSettingsIcon(topBarInstance)
-
-	rawset(menuItem, "Hover", function(self, hovering)
-		if hovering then
-			self:SetImage("rbxasset://textures/ui/Menu/HamburgerDown.png")
-		else
-			self:SetImage("rbxasset://textures/ui/Menu/Hamburger.png")
-		end
-	end)
+local function Create3DSettingsIcon(topBarInstance, panel, menubar)
+	local MenuModule = nil
+	game.CoreGui.RobloxGui.Modules:WaitForChild("Settings")
+	game.CoreGui.RobloxGui.Modules.Settings:WaitForChild("SettingsHub")
```

Comment:
All modules are loaded before corescripts run so we should not use `WaitForChild` here. If you index for it you can potentially avoid issues of waiting for a non-existent children.